src/transforms/scaling.jl

```diff
@@ -0,0 +1,58 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    Scaling(sx, sy, sz)
+
+Transform geometry or mesh by scaling each coordinate of the vertices by 
+the corresponding factor. 
+
+## Arguments
+
+- `sx`: scaling factor for the first coordinate
+- `sy`: scaling factor for the second coordinate
+- `sz`: scaling factor for the third coordinate
+
+## Examples
+
+```julia
+Scaling(1, 2, 3)
+```
+"""
+struct Scaling{T<:Real} <: GeometricTransform
+  sx::T
+  sy::T
+  sz::T
+end
+
+function Scaling(sx::T1, sy::T2, sz::T3) where {T1<:Real,T2<:Real,T3<:Real}
+  if sx <= 0 || sy <= 0 || sz <= 0
+    error("The scaling factors must be nonnegative.")
+  end
+  T = promote(sx, sy, sz)
+  Scaling(T(sx), T(sy), T(sz))
+end
+
+isrevertible(::Type{<:Scaling}) = true
+
+function preprocess(transform::Scaling, object)
+  [transform.sx, transform.sy, transform.sz]
```

Comment:
Yes, this vector is the dimension-agnostic list of scaling factors that should be a field of the ScaleCoords struct.